pkg/logql/metrics.go

```diff
@@ -133,3 +144,34 @@ func QueryType(query string) (string, error) {
 		return "", nil
 	}
 }
+
+// tagsToKeyValues converts QueryTags to form that is easy to log.
+// e.g: `Source=foo,Feature=beta` -> []interface{}{"source", "foo", "feature", "beta"}
+// so that we could log nicely!
+// If queryTags is not in canonical form then its completely ignored (e.g: `key1=value1,key2=value`)
+func tagsToKeyValues(queryTags string) []interface{} {
+	toks := strings.FieldsFunc(queryTags, func(r rune) bool {
+		return r == ','
+	})
+
+	vals := make([]string, 0)
```

Comment:
How about allocating `len(toks)`?

Reply:
Good question! Actually we can't predict the len from toks.

Why?

Because tok may or may not be in canonical form e.g (`key=val') or (`abc`). 

if all the keys are in correct form, then you will have `2*len(toks)` but if any of them is invalid we have to skip them!. Hence difficult to predict!

Reply:
Fair point.